File: sdk/typescript/src/artifact/git.ts
import { ArtifactSystem } from "../api/artifact/artifact.js";
import type { ConfigContext } from "../context.js";
import { Artifact, ArtifactSource } from "../artifact.js";
import { shell } from "./step.js";

/**
 * Builder for the Git artifact.
 *
 * Mirrors Rust `Git` struct in `sdk/rust/src/artifact/git.rs`.
 * Downloads tarball, runs configure+make to build from source.
 */
export class Git {
  async build(context: ConfigContext): Promise<string> {
    const name = "git";

    const sourceVersion = "2.53.0";

    const sourcePath = `https://sdk.vorpal.build/source/git-${sourceVersion}.tar.gz`;

    const source = new ArtifactSource(name, sourcePath).build();

    const stepScript = `mkdir -p "$VORPAL_OUTPUT/bin"

pushd ./source/${name}/git-${sourceVersion}

./configure --prefix=$VORPAL_OUTPUT

make
make install`;

    const steps = [await shell(context, [], [], stepScript, [])];

    const systems = [
      ArtifactSystem.AARCH64_DARWIN,
      ArtifactSystem.AARCH64_LINUX,
      ArtifactSystem.X8664_DARWIN,
      ArtifactSystem.X8664_LINUX,
    ];

    return new Artifact(name, steps, systems)
      .withAliases([`${name}:${sourceVersion}`])
      .withSources([source])
      .build(context);
  }
}

File: sdk/python/src/vorpal_sdk/artifact/git.py
"""Git artifact (built from source via configure+make).

Mirrors ``sdk/typescript/src/artifact/git.js``.
"""

from __future__ import annotations

from typing import TYPE_CHECKING

from vorpal_sdk.artifact import Artifact, ArtifactSource
from vorpal_sdk.step import shell
from vorpal_sdk.system import SYSTEMS

if TYPE_CHECKING:
    from vorpal_sdk.context import ConfigContext


class Git:
    """Builder for the Git artifact."""

    def build(self, context: ConfigContext) -> str:
        name = "git"
        source_version = "2.53.0"
        source_path = (
            f"https://sdk.vorpal.build/source/git-{source_version}.tar.gz"
        )

        source = ArtifactSource(name, source_path).build()

        step_script = f"""mkdir -p "$VORPAL_OUTPUT/bin"

pushd ./source/{name}/git-{source_version}

./configure --prefix=$VORPAL_OUTPUT

make
make install"""

        steps = [shell(context, [], [], step_script, [])]

        return (
            Artifact(name, steps, SYSTEMS)
            .with_aliases([f"{name}:{source_version}"])
            .with_sources([source])
            .build(context)
        )
